python/scripts/otel-instrument

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+: <<'END_DOCUMENTATION'
+`otel-instrument`
+
+This script configures and sets up OpenTelemetry Python with the values we
+expect will be used by the common AWS user. It then calls a OpenTelemetry
+Lambda script to finish initializing the OpenTelemetry Python SDK.
+
+NOTE: We renamed OpenTelemetry Lambda's script because we want this script to be
+the entry point. The layer build script renames it to
+`otel-instrument-upstream-lambda`.
+
+See more:
+https://github.com/open-telemetry/opentelemetry-lambda/blob/db6ff3a488e0d54f2f979f7540c5a5baccef5fe3/python/src/otel/otel_sdk/otel-instrument
+
+END_DOCUMENTATION
+
+export LAMBDA_RESOURCE_ATTRIBUTES="cloud.region=$AWS_REGION,cloud.provider=aws,faas.name=$AWS_LAMBDA_FUNCTION_NAME,faas.version=$AWS_LAMBDA_FUNCTION_VERSION";
+
+if [ -z ${OTEL_RESOURCE_ATTRIBUTES} ]; then
+    export OTEL_RESOURCE_ATTRIBUTES=$LAMBDA_RESOURCE_ATTRIBUTES;
+else
+    export OTEL_RESOURCE_ATTRIBUTES="$LAMBDA_RESOURCE_ATTRIBUTES,$OTEL_RESOURCE_ATTRIBUTES";
+fi
+
+# - Set `OTEL_PROPAGATORS` if not yet set, and include the xray propagator
+
+if [ -z ${OTEL_PROPAGATORS} ]; then
+    export OTEL_PROPAGATORS="tracecontext,baggage,xray";
+fi
+
+# - Call the upstream configure OTel script
+
+source /opt/otel-instrument-upstream-lambda "$@"
```

Comment:
@anuraaga My bash-fu is definitely not very strong, so I was wondering if I could ask you if this is the right command here?

This script calls the [OTel Lambda upstream otel-instrument script](https://github.com/open-telemetry/opentelemetry-lambda/blob/db6ff3a488e0d54f2f979f7540c5a5baccef5fe3/python/src/otel/otel_sdk/otel-instrument) and I took inspiration for using the `source` command from the code you linked to:

https://github.com/aws-observability/aws-otel-lambda/blob/cd6d8ba2fb8bc9b5b323dbc70f86c7529554ae7d/java/scripts/otel-handler#L13

I'm only worried about 2 things:
* Should I be using `exec` instead? Does it matter? (I guess the environment variables would stick around and that is what is important?)
* Will the arguments be sent correctly to the next script?

Thank you!

Reply:
should be no difference in this case because source is the last line.

Reply:
Yeah I think they are effectively the same here - `source` seemed to fit a bit better conceptually since it is just like the two scripts are merged into one file, which is what we're hoping for here.

As such, I don't think you need `"$@"`, I suspect it's effectively a no-op like this. The arguments will be present in `$@` without that, since we're effectively combining the two into one file.

Reply:
Thanks guys! Makes sense to me, I've made the update.